pkg/scheduler/partition.go

```diff
@@ -1099,3 +1099,15 @@ func (pc *PartitionContext) removeAllocationAsk(appID string, allocationKey stri
 		}
 	}
 }
+
+func (pc *PartitionContext) GetCurrentState() string {
+	return pc.stateMachine.Current()
+}
+
+func (pc *PartitionContext) GetStateTime() time.Time {
+	return pc.stateTime
+}
+
+func (pc *PartitionContext) GetNodeSortingPolicy() *policies.NodeSortingPolicy {
+	return pc.nodeSortingPolicy
+}
```

Comment:
I think we need to add a read lock for `GetStateTime()`, and the write lock in `handlePartitionEvent()`. because `handlePartitionEvent()` and the REST API could be called at the same time.

```
func (pc *PartitionContext) handlePartitionEvent(event objects.ObjectEvent) error {
	err := pc.stateMachine.Event(event.String(), pc.Name)
	if err == nil {
                pc.Lock()
		pc.stateTime = time.Now()
                pc.Unlock()
		return nil
	}
	// handle the same state transition not nil error (limit of fsm).
	if err.Error() == "no transition" {
		return nil
	}
	return err
}
```

`GetCurrentState()` and `GetNodeSortingPolicy()` should be safe because state machine Current() holds the lock already, and the nodeSortingPolicy only gets modified when updating the config, that is a swapping of the old object.

Reply:
Taken care.